src/main/groovy/org/labkey/gradle/task/PomFile.groovy

```diff
@@ -30,54 +29,38 @@ import org.labkey.gradle.util.BuildUtils
  */
 class PomFile extends DefaultTask
 {
-    String artifactCategory = "libs"
     Properties pomProperties = new Properties()
+    boolean isModulePom
 
     @OutputFile
     File getPomFile()
     {
-        return new File(project.buildDir, "publications/${artifactCategory}/pom-default.xml")
+        return new File(project.buildDir, "publications/${pomProperties.get("artifactCategory")}/pom-default.xml")
     }
 
     @TaskAction
     void writePomFile()
     {
             project.pom {
                 withXml {
+                    asNode().get('groupId').first().setValue(pomProperties.getProperty("groupId"))
+                    removeDependencies(asNode())
                     asNode().get('artifactId').first().setValue((String) pomProperties.getProperty("ArtifactId", project.name))
-                    // remove the tomcat dependencies with no version specified because we cannot know which version of tomcat is in use
-                    List<Node> toRemove = []
-                    def dependencies = asNode().dependencies
-                    if (!dependencies.isEmpty())
-                    {
-                        dependencies.first().each {
-                            if (it.get("groupId").first().value().first().equals("org.apache.tomcat") &&
-                                    it.get("version").isEmpty())
-                                toRemove.add(it)
-                            if (it.get('groupId').first().value().first().equals("org.labkey"))
-                            {
-                                String artifactId = it.get('artifactId').first().value().first();
-                                if (artifactId.equals("java"))
-                                    it.get('artifactId').first().setValue(['labkey-client-api'])
-                                else if (artifactId.equals("bootstrap"))
-                                    it.get('artifactId').first().setValue(ServerBootstrap.JAR_BASE_NAME)
-                            }
-                        }
-                        toRemove.each {
-                            asNode().dependencies.first().remove(it)
-                        }
-                        // FIXME it's possible to have external dependencies but no dependencies.
-                        // add in the dependencies from the external configuration as well
-                        def dependenciesNode = asNode().dependencies.first()
-                        project.configurations.api.allDependencies.each {
-                            def depNode = dependenciesNode.appendNode("dependency")
-                            depNode.appendNode("groupId", it.group)
-                            depNode.appendNode("artifactId", it.name)
-                            depNode.appendNode("version", it.version)
+                    def dependenciesNode = asNode().dependencies.first()
```

Comment:
It's been working because we've been adding an empty dependency node in `removeDependencies()` after replacing the tomcat dependencies. I changed it since if we didn't have any tomcat dependencies, we could potentially have issues